### PR TITLE
Tentative fix for issue #1634

### PR DESCRIPTION
**Pull request by xclerc**
Use `lsof` to get the number of opened files (macOS).

### CODE REVIEW

File: src/stats.ml [hunk 1]
```diff
@@ -5,9 +5,28 @@ let enabled = ref false
...
+    in
+    let prog = "/usr/sbin/lsof" in
+    let argv = [prog; "-w"; "-p"; string_of_int (Unix.getpid ())] in
+    let pid = Spawn.spawn ~prog ~argv ~stdout () in
+    Unix.close stdout;
+    match Unix.waitpid [] pid with
+    | _, Unix.WEXITED 0 ->
+      let num_lines = List.length (Io.input_lines (open_in temp)) in
+      This (num_lines - 1) (* the output contains a header line *)
+    | _ -> Unknown
```
Comment (rgrinberg) [inline src/stats.ml]: Why not use `Process.run_capture` (or one of its variations)?

**Review by rgrinberg — COMMENTED** [review]
I tested it and it works. This also deserves a CHANGES entry.

File: src/stats.ml (hunk 1 above)
Comment (xclerc) [reply src/stats.ml]: It was my first move but, if I remember correctly, it unfortunately
created a circular dependency.

**Review by rgrinberg — APPROVED** [review]

File: src/stats.ml (hunk 1 above)
Comment (xclerc) [reply src/stats.ml]: Indeed, comment added.

**Comment by xclerc** [comment]
The CI errors seem to be unrelated to this PR.

**Comment by ejgallego** [comment]
[sorry I clicked on the wrong page]

**Pull request merged**